consensus/propagation/commitment_state.go

```diff
@@ -72,32 +76,59 @@ func (p *ProposalCache) AddProposal(cb *proptypes.CompactBlock) (added bool) {
 // GetProposal returns the proposal and block for a given height and round if
 // this node has it stored or cached.
 func (p *ProposalCache) GetProposal(height int64, round int32) (*types.Proposal, *types.PartSet, bool) {
-	cb, parts, _, has := p.getAllState(height, round)
+	cb, parts, _, has := p.getAllState(height, round, true)
 	if !has {
 		return nil, nil, false
 	}
 	return &cb.Proposal, parts.Original(), has
 }
 
-func (p *ProposalCache) dumpAll() []*proposalData {
+func (p *ProposalCache) unfinishedHeights() []*proposalData {
 	p.pmtx.Lock()
 	defer p.pmtx.Unlock()
-	data := make([]*proposalData, 0, len(p.proposals))
+	data := make([]*proposalData, 0)
 	for _, heightData := range p.proposals {
+		var prop *proposalData
 		for _, pd := range heightData {
-			data = append(data, pd)
+			if prop == nil {
+				prop = pd
+				continue
+			}
+			if pd.compactBlock.Proposal.Round > prop.compactBlock.Proposal.Round {
+				prop = pd
+			}
+		}
+		if prop.block.IsComplete() {
+			continue
 		}
+		data = append(data, prop)
 	}
 	return data
 }
 
+func (p *ProposalCache) relevant(height int64, round int32) bool {
```

Comment:
what's a relevant height? maybe add a comment explaining what this method does

Reply:
https://github.com/celestiaorg/celestia-core/pull/1695/commits/4ea2f23654692c8718389178c0ab1325ebe21b79